pallets/pallet-bank/src/tests.rs

```diff
@@ -601,3 +610,46 @@ fn can_force_transfer() {
 			assert!(Bank::check_total_issuance());
 		});
 }
+
+#[test]
+fn can_check_fund_unlock_at() {
+	MockGenesisConfig::with_balances(vec![(ALICE, 1_000)]).build().execute_with(|| {
+		//
+		let unlock_block = System::block_number() + StakePeriod::get();
+		assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 100));
+
+		System::set_block_number(10);
+		let unlock_block_2 = System::block_number() + StakePeriod::get();
+		assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 200));
+
+		System::set_block_number(100);
+		let unlock_block_3 = System::block_number() + StakePeriod::get();
+		assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 300));
```

Comment:
you can insert storage directly into the storage.
Test the case where there are multiple storage items, where each contains more than 1 locks - to ensure the search works.